### PR TITLE
feat: download individual content files from context menu

### PR DESCRIPTION
**Pull request by libussa**
## Summary

- Adds a **Download** option to the file context menu in both the Content tab's tree and table views
- Uses index-based file identification (not file paths) for security
- Adds a `requireLocalAccess` security gate — requires "Local Filesystem Access" enabled on the instance
- Resolves files via both `SavePath` and `DownloadPath` from torrent properties
- Keeps the `<a>` link download approach (no blob buffering in memory) for large file support
- Hides the Download menu item entirely when local filesystem access is disabled
- Documents the feature in instance settings docs

## Test plan

- [x] Enable "Local Filesystem Access" on an instance, right-click a file in Content tab tree view → Download → file downloads via browser
- [x] Same in table view
- [x] Verify incognito mode blocks download
- [x] Verify Download option is hidden when "Local Filesystem Access" is disabled
- [x] Verify invalid file index returns 404

## Screenshot

image

Solves #970

## Summary by CodeRabbit

* **New Features**
  * Download individual files from a torrent's Content tab via a new "Download" context‑menu action in both file list and file tree. Starts a direct browser download when the instance allows local filesystem access; disabled in incognito mode.
* **Documentation**
  * Added docs describing the Content File Download feature and how to use it (right‑click a file in Content).
* **Tests**
  * Added unit tests covering download behavior, path resolution, and edge cases (permission, traversal, missing files).

### CODE REVIEW

**Comment by s0up4200** [comment]
@codex review

File: internal/api/handlers/torrents.go [hunk 1]
```diff
@@ -2077,3 +2082,167 @@ func (h *TorrentsHandler) GetDirectoryContent(w http.ResponseWriter, r *http.Req
...
+	if err != nil {
+		if respondIfInstanceDisabled(w, err, instanceID, "torrents:downloadContentFile") {
+			return
+		}
+		log.Error().Err(err).Int("instanceID", instanceID).Str("hash", hash).Msg("Failed to get torrent properties")
+		RespondError(w, http.StatusInternalServerError, "Failed to get torrent properties")
+		return
+	}
+
+	candidates := filePathCandidates(props.SavePath, props.DownloadPath, targetFileName)
```
[the remark this answers is not in the document]
Comment (libussa) [reply internal/api/handlers/torrents.go]: nice corner case, should be fixed in the latest push.  Also added a test for this

File: web/src/components/torrents/TorrentDetailsPanel.tsx [hunk 2]
```diff
@@ -89,6 +89,8 @@ export const TorrentDetailsPanel = memo(function TorrentDetailsPanel({ instanceI
   const incognitoHash = incognitoMode && torrent?.hash ? getLinuxHash(torrent.hash) : undefined
   const [pendingFileIndices, setPendingFileIndices] = useState<Set<number>>(() => new Set())
   const supportsFilePriority = capabilities?.supportsFilePriority ?? false
+  const { data: instances } = useQuery({ queryKey: ["instances"], queryFn: api.getInstances, staleTime: 60000 })
+  const hasLocalFilesystemAccess = instances?.find(i => i.id === instanceId)?.hasLocalFilesystemAccess ?? false
```
[the remark this answers is not in the document]
Comment (libussa) [reply web/src/components/torrents/TorrentDetailsPanel.tsx]: good call, fixed

**Comment by libussa** [comment]
@coderabbitai resume
